lib/src/main/java/com/cloudbees/groovy/cps/CpsTransformer.java

```diff
@@ -763,6 +763,29 @@ public void run() {
         BINARY_OP_TO_BUILDER_METHOD.put(KEYWORD_IN, "isCase");
     }
 
+    private void multipleAssignment(final Expression parentExpression,
+                                    TupleExpression tuple,
+                                    ListExpression list) {
+        List<Expression> listExpressions = list.getExpressions();
+        List<Expression> tupleExpressions = tuple.getExpressions();
+        if (listExpressions.size() < tupleExpressions.size()) {
+            // TODO: Better error handling.
+            throw new UnsupportedOperationException();
```

Comment:
Needs to at least give us a reason here -- just a simple "can't do multi-assign when variable counts differ: {0},{1}" etc etc

Reply:
See new commit.